### PR TITLE
Feat/st 662: Handle invalid Region/Hearing venue data

### PR DESCRIPTION
**Pull request by sjamihmcts**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/ST-662

### Change description ###
Handle invalid Region/Hearing venue data

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by ElifDenizEsea — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/sptribs/caseworker/event/CaseworkerRecordListing.java [hunk 1]
```diff
@@ -107,7 +109,9 @@ public AboutToStartOrSubmitResponse<CaseData, State> midEvent(CaseDetails<CaseDa
             DynamicList hearingVenueList = locationService.getHearingVenuesByRegion(regionId);
             caseData.getRecordListing().setHearingVenues(hearingVenueList);
 
-            String hearingVenueMessage = hearingVenueList.getListItems().isEmpty() ? "Unable to retrieve Hearing Venues data" : "";
+            String hearingVenueMessage = hearingVenueList == null || hearingVenueList.getListItems().isEmpty()
```
Comment (osheaf01) [inline src/main/java/uk/gov/hmcts/sptribs/caseworker/event/CaseworkerRecordListing.java:112]: Use CollectionUtils.isEmpty here?

File: src/main/java/uk/gov/hmcts/sptribs/caseworker/model/RecordListing.java [hunk 2]
```diff
@@ -183,12 +183,12 @@ public class RecordListing {
 
     @JsonIgnore
     public String getSelectedRegionVal() {
-        return this.getRegionList().getValue().getLabel();
+        return this.getRegionList() != null ? this.getRegionList().getValue().getLabel() : null;
     }
```
Comment (osheaf01) [inline src/main/java/uk/gov/hmcts/sptribs/caseworker/model/RecordListing.java:187]: Could use Objects.nonNull here? Also, for consistency, if you're null checking `this.getRegionList()`, should you not null check `this.getRegionList().getValue()`?

**Review by osheaf01 — CHANGES_REQUESTED** [review]
Just a couple of comments?

File: src/main/java/uk/gov/hmcts/sptribs/caseworker/event/CaseworkerRecordListing.java (hunk 1 above)
Comment (sjamihmcts) [reply src/main/java/uk/gov/hmcts/sptribs/caseworker/event/CaseworkerRecordListing.java:112]: hearingVenueList would never be empty so only not null check would be sufficient.

File: src/main/java/uk/gov/hmcts/sptribs/caseworker/model/RecordListing.java (hunk 2 above)
Comment (sjamihmcts) [reply src/main/java/uk/gov/hmcts/sptribs/caseworker/model/RecordListing.java:187]: this.getRegionList().getValue() can never be null too. We are giving default label as expected by DynamicList. So this check is not really needed.

**Pull request merged**